### PR TITLE
Add timeouts to recycler, other improvements

### PR DESCRIPTION
**Pull request by hansmi**

### CODE REVIEW

File: recycler.sh [hunk 1]
```diff
@@ -339,10 +332,8 @@ recycle_volume() {
 
       echo "Annotating ${vol_name} as failed at ${failed_at}"
 
-      patch_result=$(api_call PATCH "/api/v1/persistentvolumes/${vol_name}" "$patch" application/strategic-merge-patch+json)
-      if [[ "$?" != 0 ]]; then
-        echo "Couldn't annotate ${vol_name} with failed timestamp. The response was:"
-        echo "$patch_result"
+      if ! patch_result=$(api_call PATCH "/api/v1/persistentvolumes/${vol_name}" "$patch" application/strategic-merge-patch+json); then
```
Comment (mhutter) [inline recycler.sh]: either re-add `echo` or get rid of `patch_result` entirely

**Review by mhutter — CHANGES_REQUESTED** [review]

**Comment by mhutter** [comment]
Oh and there is a tiny typo in the last commit message (`s/os/is`)...

**Comment by hansmi** [comment]
> Oh and there is a tiny typo in the last commit message (s/os/is)...

I'm sorry, but I can't find it. Can you give me the commit ID?

File: recycler.sh (hunk 1 above)
Comment (hansmi) [reply recycler.sh]: Done in the latest revision.

**Comment by mhutter** [comment]
6f996e0b396f5473ce2b46a6c2513df7441c9c49, 2nd paragraph, 6th/7th word

**Comment by hansmi** [comment]
> 6f996e0, 2nd paragraph, 6th/7th word

Oh, it's `s/us/is/`, not `s/os/is`. FIxed in latest revision.

**Comment by mhutter** [comment]
Typos in the typo nitpick ✅

**Review by mhutter — APPROVED** [review]

**Pull request merged**